app/graphql/mutations/update_squeak.rb

```diff
@@ -3,22 +3,33 @@ class UpdateSqueak < Mutations::BaseMutation
     argument :id, ID, required: true
     argument :report, Boolean, required: false
     argument :nut, Boolean, required: false
+    argument :approved, Boolean, required: false
 
     field :squeak, Types::SqueakType, null: false
 
-    def resolve(id: , report: false, nut: false)
+    def resolve(id: , report: false, nut: false, approved: nil)
       squeak = Squeak.find(id)
       if squeak
-        if report == true
-          squeak.reports += 1
-        elsif nut == true
-          squeak.nuts += 1
-        end
-        squeak.save
+        check_arguments(squeak, report, nut, approved)
       end
       { squeak: squeak }
       rescue ActiveRecord::RecordNotFound => e
         GraphQL::ExecutionError.new(e)
     end
+
+    def check_arguments(squeak, report, nut, approved)
```

Comment:
Nice refactoring! This is much more readable than the longer conditional. 

This is a nit, but I wonder if `perform_update` or something could be a better name for this function. I was thinking it was going be parameter validation based on the name, but it is actually changing the state of the squeak